src/components/ArticlePage.jsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { format } from "date-fns";
+import { fetchSingleArticle } from "../../utils";
+import "../css/ArticlePage.css";
+
+const ArticlePage = () => {
+  const { article_id } = useParams();
+  const [currArticle, setCurrArticle] = useState({});
+
+  useEffect(() => {
+    fetchSingleArticle(article_id)
+      .then((article) => {
+        setCurrArticle(article);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, []);
+
+  if (Object.keys(currArticle).length === 0) return null;
+
+  const { title, author, topic, created_at, votes, article_img_url, body } =
+    currArticle;
```

Comment:
We should give feedback to the user that the article is loading while we fetch the data 🙂